public/apps/login/login-page.tsx

```diff
@@ -49,16 +50,34 @@ interface LoginButtonConfig {
   buttonstyle: string;
 }
 
-function redirect(serverBasePath: string) {
-  // navigate to nextUrl
+export function getNextPath(serverBasePath: string) {
   const urlParams = new URLSearchParams(window.location.search);
   let nextUrl = urlParams.get('nextUrl');
   if (!nextUrl || nextUrl.toLowerCase().includes('//')) {
     // Appending the next url with trailing slash. We do so because in case the serverBasePath is empty, we can simply
     // redirect to '/'.
     nextUrl = serverBasePath + '/';
   }
-  window.location.href = nextUrl + window.location.hash;
+  const savedTenant = getSavedTenant();
+  const url = new URL(
+    window.location.protocol + '//' + window.location.host + nextUrl + window.location.hash
```

Comment:
I would consider extracting these and the like to a constants/util file. It seems like we can use generators and constants to avoid the duplicative hard coding.

Reply:
What is duplicated here? 

This constructs the nextUrl based off the nextUrl param from the login route and will add the tenant info if there is a tenant saved in local storage.

Reply:
Opting not to address this comment. Please leave a suggestion of how you think this can be organized.